apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsLinkControl.tsx

```diff
@@ -89,24 +230,87 @@ const RadioCard = ({ children, ...rest }: PropsWithChildren<UseRadioProps>) => {
 
 interface RadioContentProps {
   selectedLinkType: string
+  description?: string
   data: string
   handleChange: (value: string) => void
 }
 
 const RadioContent = ({
   selectedLinkType,
+  description,
   data,
   handleChange,
 }: RadioContentProps): JSX.Element => {
+  const router = useRouter()
+  const siteId = String(router.query.siteId)
+  const {
+    isOpen: isPageLinkModalOpen,
+    onOpen: onPageLinkModalOpen,
+    onClose: onPageLinkModalClose,
+  } = useDisclosure()
+  const potentialInternalLinkResourceId = getResourceIdFromReferenceLink(data)
+  const { data: potentialInternalLinkResource } =
+    trpc.resource.getWithFullPermalink.useQuery(
+      {
+        resourceId: potentialInternalLinkResourceId,
+      },
+      {
+        enabled: potentialInternalLinkResourceId !== "",
```

Comment:
nit: might want to guard against all falsey values (if it's guaranteed to be a string, number will miss `0` but i assume that's not valid)

```suggestion
        enabled: !!potentialInternalLinkResourceId,
```

Reply:
Updated in 7f139851126b5ab4f72d99fbbf9c5588cb2f9adc.